int16 err_x
int16 err_y
